packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,237 @@
+  const phoneNumberLength = phoneNumber.length;
+
+  // we need to return the value with no formatting if its less then four digits
+  // this is to avoid weird behavior that occurs if you  format the area code to early
+  // if phoneNumberLength is greater than 10 we don't do any formatting
+
+  if (phoneNumberLength < 4 || phoneNumberLength > 10) {
+    return phoneNumber;
+  }
+
+  // if phoneNumberLength is greater than 4 and less the 7 we start to return
+  // the formatted number
+  if (phoneNumberLength < 7) {
+    return `(${phoneNumber.slice(0, 3)})${phoneNumber.slice(3)}`;
+  }
+
+  // finally, if the phoneNumberLength is greater then seven, we add the last
+  // bit of formatting and return it.
+  return `(${phoneNumber.slice(0, 3)})${phoneNumber.slice(3, 6)}-${phoneNumber.slice(6, phoneNumber.length)}`;
+};
+
+const DialpadContainer = (props: {
+  errorText: string;
+  defaultText: string;
+  containerStyles?: IStyle;
+  rowStyles?: IStyle;
+  textFieldStyles?: Partial<ITextFieldStyles>;
+  digitStyles?: IStyle;
+  subStyles?: IStyle;
+}): JSX.Element => {
+  const theme = useTheme();
+  const [textValue, setTextValue] = useState(props.defaultText);
+  const [error, setError] = useState('');
+  const onClickDialpad = (input: string): void => {
+    setError('');
+    if (textValue === props.defaultText) {
+      setTextValue(input);
+    } else {
+      setTextValue(textValue + input);
+    }
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const setText = (e: any): void => {
+    // if text content includes non-valid dialpad input return an error
+    if (!`${e.target.value}`.match(/^[0-9*#+()-]*$/)) {
+      setError(props.errorText);
+    } else {
+      setError('');
+      setTextValue(e.target.value);
+    }
+  };
+
+  return (
+    <div className={mergeStyles(containerStyles(theme), props.containerStyles)}>
+      <TextField
+        styles={concatStyleSets(textFieldStyles(theme), props.textFieldStyles)}
+        value={formatPhoneNumber(textValue, props.defaultText)}
+        onChange={setText}
+        errorMessage={error}
+        onClick={() => {
+          if (textValue === props.defaultText) {
+            setTextValue('');
+          }
+        }}
+        onFocus={() => {
+          if (textValue === props.defaultText) {
+            setTextValue('');
+          }
+        }}
+      />
+      <div className={mergeStyles(rowStyles(theme), props.rowStyles)}>
+        <button
+          className={mergeStyles(digitStyles(theme), props.digitStyles)}
+          onClick={() => {
+            onClickDialpad('1');
+          }}
+        >
+          1
+        </button>
+        <button
+          className={mergeStyles(digitStyles(theme), props.digitStyles)}
+          onClick={() => {
+            onClickDialpad('2');
+          }}
+        >
+          2<div className={mergeStyles(subStyles(theme), props.subStyles)}>ABC</div>
+        </button>
+        <button
```

Comment:
There's a lot of duplication here, can we have a `DialPadButton` that encapsulates the styling etc. so each button becomes:
```
<DialPadButton primaryContent={3} secondaryContent={DEF}>
```
(can change the names of those props)

Reply:
nit: also consider programmatically dictating these, something like:
```
const dialPadButtons = [
    [
      { primaryContent: 1 },
      { primaryContent: 2, secondaryContent: 'ABC' },
      { primaryContent: 3, secondaryContent: 'DEF' }
    ],
    [
      { primaryContent: 1}, // change for actual, correct info
      { primaryContent: 2, secondaryContent: 'ABC' },
      { primaryContent: 3, secondaryContent: 'DEF' }
    ],
    [
      { primaryContent: 1},
      { primaryContent: 2, secondaryContent: 'ABC' },
      { primaryContent: 3, secondaryContent: 'DEF' }
    ],
    [
      { primaryContent: 1 },
      { primaryContent: 2, secondaryContent: 'ABC' },
      { primaryContent: 3, secondaryContent: 'DEF' }
    ]
  ];
<Dialpad buttons={dialPadButtons}
```

Reply:
updated